### PR TITLE
Handle weirdness of forking private repos into org

### PR DESCRIPTION
**Pull request by bloodearnest**
opensafely/vaccine-eligibility was forked into opensafely from a personal private repo. Because of github rules, as it's not a paid for repo, we cannot control branch protection so this 403s.

This change handles that more gracefully. In future, we want to avoid forking private repos into the org.

Also:
 - update to the config
 - fix bug in applying branch permissions

### CODE REVIEW

File: manage-github.py [hunk 1]
```diff
@@ -74,9 +74,17 @@ def protect_branch(repo, branch=None, **kwargs):
             current_protection = convert_protection(protected_branch.get_protection())
         except GithubException as e:
             if e.status == 404:
+                # new repo no protection set
                 protection = kwargs
             else:
-                raise
+                yield client.Change(
```
Comment (madwort) [inline manage-github.py:80]: might be worth adding a comment here, equivalent to line 77, saying e.g. `# private repo forked into our org` ?

**Review by madwort — COMMENTED** [review]
@bloodearnest looks good, looks like it does what it says on the tin, I’m slightly unsure about the higher level question of - what are we expecting to do with these repos that we can’t enforce branch protection on, just leave them with whatever settings they have already? maybe if you could add this rationale to the headline PR comment for future reference?

File: manage-github.py (hunk 1 above)
Comment (bloodearnest) [reply manage-github.py:80]: good call

File: manage-github.py [hunk 2]
```diff
@@ -77,9 +77,11 @@ def protect_branch(repo, branch=None, **kwargs):
                 # new repo no protection set
                 protection = kwargs
             else:
+                # this occurs when a private repo is forked *into* the opensafely org
+                # currently just vaccine-eligibility repo, we want to avoid that in future.
                 yield client.Change(
                     lambda: None,
-		    'ERROR: exception getting branch protection on {}/{}:\n{}',
```
Comment (madwort) [inline manage-github.py]: good to remove the `:` but don't we still need the `,`?

File: manage-github.py (hunk 1 above)
Comment (bloodearnest) [reply manage-github.py:82]: added

File: manage-github.py (hunk 2 above)
Comment (bloodearnest) [reply manage-github.py]: yep, tests caught that failure \o/ fixing

**Review by madwort — APPROVED** [review]

**Pull request merged**